spec/invoice_item_repository_spec.rb

```diff
@@ -0,0 +1,163 @@
+                                                      :unit_price       => BigDecimal(10.99,4),
+                                                      :updated_at   => Time.now.to_s
+                                                      })
+
+      invoice_item_2 = invoice_item_repository.create({
+                                                      :id           => 89101112,
+                                                      :item_id  => 456,
+                                                      :invoice_id  => 13141516,
+                                                      :created_at   => Time.now.to_s,
+                                                      :quantity       => 4,
+                                                      :unit_price       => BigDecimal(15.00,4),
+                                                      :updated_at   => Time.now.to_s
+                                                      })
+
+      expect(invoice_item_repository.find_all_by_invoice_id(invoice_item_1.invoice_id)).to eq([invoice_item_1, invoice_item_2])
+    end
+
+    it 'can update an invoice item' do
+      invoice_item_repository = @sales_engine.invoice_items
+      invoice_item_1 = invoice_item_repository.create({
+                                                      :id           => 1234567,
+                                                      :item_id  => 89101112,
+                                                      :invoice_id  => 13141516,
+                                                      :created_at   => Time.now.to_s,
+                                                      :quantity       => 4,
+                                                      :unit_price       => BigDecimal(10.99,4),
+                                                      :updated_at   => Time.now.to_s
+                                                      })
+
+      invoice_item_2 = invoice_item_repository.create({
+                                                      :id           => 89101112,
+                                                      :item_id  => 456,
+                                                      :invoice_id  => 13141516,
+                                                      :created_at   => Time.now.to_s,
+                                                      :quantity       => 4,
+                                                      :unit_price       => BigDecimal(15.00,4),
+                                                      :updated_at   => Time.now.to_s
+                                                      })
+
+      attributes = {quantity: 25}
+      expected = invoice_item_repository.update(invoice_item_1.quantity, attributes)
+
+      expect(expected.quantity).to eq(25)
+    end
+
+    it 'can delete invoice item' do
+      invoice_item_repository = @sales_engine.invoice_items
+      invoice_item_1 = invoice_item_repository.create({
+                                                      :id           => 1234567,
```

Comment:
do we need to set the id? the create method creates an id using the last id + 1... same with the invoice_item_2.